sage-ios/SAGE/SAGE/AddSchoolController.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  AddSchoolController.swift
+//  SAGE
+//
+//  Created by Sameera Vemulapalli on 11/21/15.
+//  Copyright © 2015 Cal Blueprint. All rights reserved.
+//
+
+import UIKit
+
+class AddSchoolController: UIViewController {
+    
+    var director: User?
+    var place: GMSPlace?
+    var currentErrorMessage: ErrorView?
+
+    override func viewDidLoad() {
+        let addSchoolView = AddSchoolView(frame: self.view.frame)
+        self.view = addSchoolView
+        self.title = "Add School"
+        self.navigationItem.rightBarButtonItem = UIBarButtonItem(title: "Finish", style: .Done, target: self, action: "completeForm")
+        addSchoolView.location.button.addTarget(self, action: "locationButtonTapped", forControlEvents: .TouchUpInside)
+        addSchoolView.director.button.addTarget(self, action: "directorButtonTapped", forControlEvents: .TouchUpInside)
+    }
+    
+    func locationButtonTapped() {
+        let tableViewController = AddSchoolLocationTableViewController()
+        if let topItem = self.navigationController!.navigationBar.topItem {
+            topItem.backBarButtonItem = UIBarButtonItem(title: "", style: .Plain, target: nil, action: nil)
+        }
+        self.navigationController!.pushViewController(tableViewController, animated: true)
+    }
+    
+    func directorButtonTapped() {
+        let tableViewController = AddSchoolDirectorTableViewController()
+        if let topItem = self.navigationController!.navigationBar.topItem {
+            topItem.backBarButtonItem = UIBarButtonItem(title: "", style: .Plain, target: nil, action: nil)
+        }
+        self.navigationController!.pushViewController(tableViewController, animated: true)
+    }
+    
+    @objc private func completeForm() {
+        let addSchoolView = (self.view as! AddSchoolView)
+        if addSchoolView.name.textField.text == nil || addSchoolView.name.textField.text == "" {
+            self.showErrorAndSetMessage("What's the school's name?", size: 64.0)
```

Comment:
i refactored errorview and stuff so that its uicolor.maincolor and you don't have to pass in a height anymore but it's covering the top form field/doesn't look great
should i make it less transparent or should i just use an alertcontroller